src/form/components/Dropdown2/DropdowItem.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import List from '../../../components/List';
+import useDropdowContext from './useDrodpownContext';
+
+function DropdownItem({
+  children,
+  value,
+}) {
+  const { onChange, value: dropdownValue } = useDropdowContext();
+  return (
+    <List.Item
+      {...value === dropdownValue && { className: 'tyk-list__item--selected' }}
```

Comment:
`List.Item` should probably have a `selected` prop. The dropdown component should not be responsible of setting List specific classes.